### PR TITLE
Ocd 2634

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: PENDING_RELEASE_NOTES.md [hunk 1]
```diff
@@ -3,7 +3,12 @@
 _Date TBD_
 
 ### New Features
+* Added ROLE_ONC user role. ROLE_ONC replaces the existing ROLE_ADMIN and all relevant accounts were converted to the new role.
```
Comment (andlar) [inline PENDING_RELEASE_NOTES.md]: Weren't these first few new lines already deployed to PROD?

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/scheduler/GetAllActionPermissions.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+            }
+        } catch (Exception e) {
+            LOGGER.error(e);
+            return false;
+        }
+    }
+
+    private Boolean doesUserHavePermissionToJob(final ChplJob job) {
+        //Get the authorities from the job
+        if (job.getGroup() != null && job.getGroup().equals("chplJobs")) {
```
Comment (andlar) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/scheduler/GetAllActionPermissions.java]: This method is only called from the `hasAccess` method, right? And from that method, only after we've already determined that the group == "chplJobs", so this if isn't needed, correct?

Comment (tmy1313) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/permissions/domains/scheduler/GetAllActionPermissions.java]: True - change made.

File: PENDING_RELEASE_NOTES.md (hunk 1 above)
Comment (tmy1313) [reply PENDING_RELEASE_NOTES.md]: Update notes

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/schedule/ChplOneTimeTrigger.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package gov.healthit.chpl.domain.schedule;
+
+import java.io.Serializable;
+
+public class ChplOneTimeTrigger implements Serializable {
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/domain/schedule/ChplOneTimeTrigger.java]: Why doesn't this class need a name or group like the ChplTrigger class has? Purely based on the class names, it looks like it should be more of a hierarchy with parent class ChplTrigger and subclasses RepeatableChplTrigger and OneTimeChplTrigger or something like that. Would any refactoring like that be appropriate here or is it not really the right thing to do? I'm not as familiar with the objects as you are.

File: chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/SchedulerManagerImpl.java [hunk 4]
```diff
@@ -166,20 +197,34 @@ public ChplTrigger updateTrigger(final ChplTrigger trigger) throws SchedulerExce
...
-                }
+            for (JobKey jobKey : scheduler.getJobKeys(groupEquals(group))) {
+                JobDetail jobDetail = scheduler.getJobDetail(jobKey);
+                jobs.add(new ChplJob(jobDetail));
+            }
+        }
+
+        //TODO: There is a problem using @PostFilter - some sort of ACL issue
+        //Until that is figured out...
+        Iterator<ChplJob> iterator = jobs.iterator();
```
Comment (kekey1) [inline chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/SchedulerManagerImpl.java]: I thought you had a problem with PostFilter for the surveillance stuff but figured it out. Am I remembering that wrong or is this a different problem?

Comment (tmy1313) [reply chpl/chpl-service/src/main/java/gov/healthit/chpl/manager/impl/SchedulerManagerImpl.java]: I completely forgot!  I'll update this to work with @PostFilter.

**Pull request merged**